src/rws_interface.cpp

```diff
@@ -117,6 +117,111 @@ std::string RWSInterface::getIOSignal(const std::string& iosignal)
   return result;
 }
 
+bool RWSInterface::getMechanicalUnitStaticInfo(const std::string& mechunit, MechanicalUnitStaticInfo* p_static_info)
+{
+  bool result = false;
+
+  if (p_static_info)
+  {
+    RWSClient::RWSResult rws_result = rws_client_.getMechanicalUnitStaticInfo(mechunit);
+
+    if(rws_result.success)
+    {
+      p_static_info->task_name = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "task-name"));
+      p_static_info->is_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                             XMLAttribute("class", "is-integrated-unit"));
+      p_static_info->has_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                              XMLAttribute("class", "has-integrated-unit"));
+
+      std::string type = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "type"));
+      if(type == "None")
+      {
+        p_static_info->type = NONE;
+      }
+      else if(type == "TCPRobot")
+      {
+        p_static_info->type = TCP_ROBOT;
+      }
+      else if(type == "Robot")
+      {
+        p_static_info->type = ROBOT;
+      }
+      else if(type == "Single")
+      {
+        p_static_info->type = SINGLE;
+      }
+      else
+      {
+        p_static_info->type = UNDEFINED;
+      }
```

Comment:
Why the different approach here, compared to #106?

Reply:
Don't remember, different days doing the implementations I guess 😅

Reply:
Changed to be similar to #106 in b38e52c